### PR TITLE
Add PNG source

### LINKED ISSUE #510: PNG Source for ClientSDK
Implement PNG source.

### PR DESCRIPTION
**Pull request by abramov-oleg**

### CODE REVIEW

File: savant/client/utils/img_header_parse.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from savant.gstreamer.codecs import Codec
+
+# positive lookbehind is included to avoid matching density in the JPEG header
+PATTERN = re.compile(r'(?<=, )(?P<width>\d+)( x |x)(?P<height>\d+)')
+
+
+def get_image_size_codec(file: Union[str, PathLike, BinaryIO]) -> Tuple[int, int, str]:
+    """Get JPEG or PNG image width and height by parsing the file header.
+    :param file: Path to an image file or a file handle to an image file opened as binary.
+    :return: Image width and height.
```
Comment (tomskikh) [inline savant/client/utils/img_header_parse.py]:
```suggestion
    :return: Image width, height and codec.
```

File: savant/client/frame_source/image_source.py [hunk 2]
```diff
@@ -148,8 +152,30 @@ def _update_param(self, name, value) -> 'JpegSource':
...
         return (
-            f'JpegSource('
+            f'{self.__class__.__name__}('
             f'source_id={self._source_id}, '
             f'file={self._file}, '
             f'pts={self._pts})'
         )
+
+
+class JpegSource(ImageSource):
```
Comment (tomskikh) [inline savant/client/frame_source/image_source.py:162]: Do we need to have different classes for JPEG and PNG sources? They are the same anyway and there's no validation the file type matches the class.

**Review by tomskikh — CHANGES_REQUESTED** [review]
Update docs for savant.client.

File: savant/client/frame_source/image_source.py (hunk 2 above)
Comment (bwsw) [reply savant/client/frame_source/image_source.py:162]: @tomskikh the idea is to make the use self-explanatory. Please keep as is.

**Comment by abramov-oleg** [comment]
> Update docs for savant.client.

Added Png sources to API ref, updated supported sources note on the Client SDK page.

**Review by tomskikh — APPROVED** [review]

**Pull request merged**